### PR TITLE
New board variant (3D printer controller): PRNTR_F407

### PR DESCRIPTION
**Pull request by ghent360**
<second time is the charm. I don't know why the previous PR keeps closing on me>
Good sir,

I have this 3D printer controller project (https://github.com/ghent360/PrntrBoard). Initially I used NUCLEO-64 board (F446) to host the MCU. However recently I build a replacement for the NUCLEO-64 "brain" - similar form factor, but using F407VE chip which has a bit more GPIO pins.

This PR add the board variant files. It creatively copies from the DISCO_F407VG as well as the ARMED_V1 variant.

The board schematics is in KiCAD format - open the Nucleo-64-407.pro project, then the Nucleo-64-407.sch schematics file.

### CODE REVIEW

**Comment by fpistm** [comment]
I guess this is because you delete the base commit of the PR or something like this.

**Comment by ghent360** [comment]
I'll admit my git skills are rudimentary at best.

File: variants/PRNTR_F407_V1/variant.h [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+#define EXP4_4  PB5  // SPI_MOSI
+#define EXP4_5  PC1  // !EN_MOTORS
+#define EXP4_6  PB3  // SPI_CLK
+//      EXP4_7  12-24V
+//      EXP4_8  3.3V
+//      EXP4_9  GND
+//      EXP4_10 5V
+
+// This must be a literal
+#define NUM_DIGITAL_PINS        80
```
Comment (fpistm) [inline variants/PRNTR_F407_V1/variant.h]: Only one comment about NUM_DIGITAL_PINS.
Analog pin definition should be count as they are also digital pins.
`digitalRead(A0)` will not be functional as macro
`#define digitalPinToPinName(p)      (((uint32_t)p < NUM_DIGITAL_PINS) ? digitalPin[p] : NC)`
will return `NC`.

Comment (fpistm) [reply variants/PRNTR_F407_V1/variant.h]: No more relevant as duplicated analog pins removed:
https://github.com/stm32duino/Arduino_Core_STM32/commit/689a18712feda63bcba6078d30dcdb84bb0f2ba9

**Comment by ghent360** [comment]
Thank you. I'll check it soon.

**Pull request merged**